[EPMEDU-2611] Update route icons and solve route overlay

vladandreip opened the pull request:
Update Icon for current user location
Solve destination icon
Solve route overlay

vladandreip committed:
[EPMEDU-2611]
Update Icon for current user location
Solve destination icon
Solve route overlay

Fileek commented on feature/tabsflow/home/src/main/java/com/epmedu/animeal/home/presentation/ui/map/RouteView.kt:
```diff
@@ -115,4 +124,25 @@ private fun fetchRoute(
+    state: HomeState
+): MapBoxRouteInitOptions {
+    val routeLineResources = RouteLineResources.Builder()
+        .originWaypointIcon(R.drawable.ic_your_location)
```
This icon should be set for user location when there is no route, please see [the design](https://www.figma.com/file/ihPSDMVPPlB6lIstYERxg5/Doggo?type=design&node-id=1776-1545&mode=dev)

vladandreip replied:
Got it! Thank you for the observation! I got it refactored :)

Fileek commented on feature/tabsflow/home/src/main/java/com/epmedu/animeal/home/presentation/ui/map/RouteView.kt:
```diff
@@ -115,4 +124,25 @@ private fun fetchRoute(
+        .originWaypointIcon(R.drawable.ic_your_location)
+
+    state.feedState.feedPoint?.getDrawableRes()?.let { feedingPointDrawable ->
+        routeLineResources.destinationWaypointIcon(feedingPointDrawable)
```
This icon is too big and it has wrong color, try to set it from `state.feedingPointState.currentFeedingPoint` instead

vladandreip replied:
Hey @Fileek 

I changed the icon feed from `state.feedingPointState.currentFeedingPoint`, however, there is a problem retrieving the image when the user closes the application and opens it again. 

When closes/reopens the application `state.feedingPointState.currentFeedingPoint` returns a `null `icon, resulting in the following map rendering: 

![Screenshot_20230828_150034](https://github.com/AnimealProject/animeal_android/assets/28179406/c5dc952d-c34a-4555-a649-5978b2c08153)

Fileek replied:
Let's then use `feedPoint` from `feedState`, but update its `feedStatus` in the `DefaultFeedingHandler` depending on operation.

Fileek replied:
And please solve the scaling issue:
<details>
<summary>Screenshots:</summary>

| Without route  | With route |
| ------------- | ------------- |
| ![image](https://github.com/AnimealProject/animeal_android/assets/83027107/13c79386-7ede-445a-bee4-1e540004cf18)  | ![image](https://github.com/AnimealProject/animeal_android/assets/83027107/c9eec6bd-5dbc-49b1-b7dd-6c544816d922)  |
</details>

vladandreip replied:
@Fileek Unfortunately the API scales the image automatically when setting the `destinationWaypointIcon`
I also tried XML's representing animal states with a smaller dimension but the library scales them to the same dimension even though they are smaller. 

I also checked the Mapbox API for a method to scale the `destinationWaypointIcon`, but they do not permit it :(

Fileek replied:
The icon is not clickable now. It should be clickable to finish feeding.

vladandreip committed:
[EPMEDU-2611]
Update Icon for current user location (no route established)
Small refactor

Fileek reviewed (commented):
Is it possible to remove the arrow (marked on the screenshot below), but not disable the icon rotation?
<details>
<summary>Screenshot</summary>

![image](https://github.com/AnimealProject/animeal_android/assets/83027107/a0d65786-57a4-49f7-8f5e-ea8f50d5ba9e)
</details>

Fileek commented on feature/tabsflow/home/src/main/java/com/epmedu/animeal/home/presentation/ui/map/RouteView.kt:
```diff
@@ -12,23 +15,32 @@ import com.epmedu.animeal.router.model.RouteResult
-            MapboxRouteLineOptions.Builder(mapView.context).build()
-        )
-    )
+    mapView.setMapViewLocationPuck(mapView.context)
```
This call shouldn't be there, but rather in the above composable `MapboxMap`

Fileek commented on feature/tabsflow/home/src/main/java/com/epmedu/animeal/home/presentation/ui/map/RouteView.kt:
```diff
@@ -12,23 +15,32 @@ import com.epmedu.animeal.router.model.RouteResult
-            MapboxRouteLineOptions.Builder(mapView.context).build()
-        )
-    )
+    mapView.setMapViewLocationPuck(mapView.context)
```
It doesn't make much sense to pass context, since you can retrieve it from MapView inside the function.

vladandreip replied:
that is correct, I had an incomplete refactor

vladandreip committed:
[EPMEDU-2611]
FetchCurrentFeeding point after feeding started -> updates the status of the feedingPoint

vladandreip committed:
[EPMEDU-2611]
Solve feeding point overlap for destination icon

Fileek committed:
Merge branch 'develop' into bug/EPMEDU-2611_Route_Icons_And_Overlap

Fileek committed:
Fix your location icon

Fileek committed:
Rollback destination marker changes

Fileek committed:
Remove redundant Appcompat library

Fileek committed:
Move location puck setting to map view initialisation

Fileek committed:
Fix detekt

Fileek committed:
Fix route overlay and color

Fileek committed:
Replace overlay fix workaround with the proper solution

Fileek commented:
PR is finished. Replacing route shape does not look feasible so far, but I changed the color.
![Screenshot_20240103-014529_Animeal Dev.jpg](https://github.com/AnimealProject/animeal_android/assets/83027107/467e3e06-008a-4ae5-8378-7c642d6af0f7)

ozkarram commented on feature/tabsflow/home/src/main/java/com/epmedu/animeal/home/presentation/ui/map/MapView.kt:
```diff
@@ -42,6 +45,7 @@ import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.channels.BufferOverflow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.debounce
+import com.mapbox.maps.plugin.locationcomponent.R as LocationComponentR
```
Please copy this to our own drawable folder and import it from there, this icon could change anytime in the future

Fileek replied:
Done

Commit:
Copy shadow icon for user location into project resources
